### PR TITLE
SC MXNet to TF Regression Fix

### LINKED ISSUE #2688: Sound classifier accuracy with TF much less than with MXNet (Mac, CPU).
Using the following test code: 

```
import turicreate as tc
from os.path import basename

# Load the audio data and meta data.
data = tc.load_audio("/Users/hoytak/data/ESC-50/audio/")
meta_data = tc.SFrame.read_csv("~/data/ESC-50/meta/esc50.csv")

data['filename'] = data['path'].apply(lambda p: basename(p))
data = data.join(meta_data)

# Drop all records which are not part of the ESC-10.
data = data.filter_by('True', 'esc10')

# Make a train-test split, just use the first fold as our test set.
test_set = data.filter_by(1, 'fold')
train_set = data.filter_by(1, 'fold', exclude=True)

model = tc.sound_classifier.create(train_set, target='category', feature='audio')
```

On 5.8, 4 subsequent runs produced training accuracies 0.913, 0.923, 0.919, and 0.917.  With change 41ccb592f9e9ded4298662a697c729f2f62dfe0f, the accuracies were 0.856, 0.847, 0.863, 0.823.  In addition, the accuracy frequently went up, then down in the training.  It's likely the step size is set too large. 

Example runs: 
```
| 8                       | 0.820                   | 0.775                   | 69.238                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.838                   | 0.838                   | 69.567                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.823                   | 0.762                   | 69.883                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
```

```
+-------------------------+-------------------------+-------------------------+-------------------------+
| 8                       | 0.873                   | 0.787                   | 67.856                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.826                   | 0.787                   | 68.174                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.863                   | 0.800                   | 68.511                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
```

### PR DESCRIPTION
**Pull request by fareeha**
Closes #2688 

1) Softmax was being applied twice: `self.predictions` which had softmax activation applied on it was being passed to `tf.nn.softmax_cross_entropy_with_logits_v2` to compute loss. Fixed this by passing last layer activations to `softmax_cross_entropy_with_logits_v2` instead.

2) Optimizer for TF changed from Adam to NAG to match MXNET. 

**ESC-10 on MXNet 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.372             | 451.111           |
+-------------------+-------------------+-------------------+
| 2                 | 0.592             | 451.459           |
+-------------------+-------------------+-------------------+
| 3                 | 0.675             | 451.871           |
+-------------------+-------------------+-------------------+
| 4                 | 0.782             | 452.249           |
+-------------------+-------------------+-------------------+
| 5                 | 0.819             | 452.664           |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 453.045           |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 453.435           |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 453.875           |
+-------------------+-------------------+-------------------+
| 9                 | 0.908             | 454.272           |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 454.647           |
+-------------------+-------------------+-------------------+
```

**ESC-10 on TF 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.163             | 317.240           |
+-------------------+-------------------+-------------------+
| 2                 | 0.708             | 317.564           |
+-------------------+-------------------+-------------------+
| 3                 | 0.746             | 317.916           |
+-------------------+-------------------+-------------------+
| 4                 | 0.772             | 318.248           |
+-------------------+-------------------+-------------------+
| 5                 | 0.801             | 318.589           |
+-------------------+-------------------+-------------------+
| 6                 | 0.826             | 318.936           |
+-------------------+-------------------+-------------------+
| 7                 | 0.850             | 319.254           |
+-------------------+-------------------+-------------------+
| 8                 | 0.878             | 319.577           |
+-------------------+-------------------+-------------------+
| 9                 | 0.893             | 319.916           |
+-------------------+-------------------+-------------------+
| 10                | 0.905             | 320.255           |
+-------------------+-------------------+-------------------+
```

**Using pre-computed ESC-10 deep features on MXNET 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.371             | 161.865           |
+-------------------+-------------------+-------------------+
| 2                 | 0.591             | 162.535           |
+-------------------+-------------------+-------------------+
| 3                 | 0.677             | 163.186           |
+-------------------+-------------------+-------------------+
| 4                 | 0.781             | 163.839           |
+-------------------+-------------------+-------------------+
| 5                 | 0.820             | 164.504           |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 165.188           |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 165.874           |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 166.535           |
+-------------------+-------------------+-------------------+
| 9                 | 0.900             | 167.130           |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 167.723           |
+-------------------+-------------------+-------------------+
```

**Using pre-computed ESC-10 deep features on TF 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.347             | 257.007           |
+-------------------+-------------------+-------------------+
| 2                 | 0.605             | 257.468           |
+-------------------+-------------------+-------------------+
| 3                 | 0.674             | 257.904           |
+-------------------+-------------------+-------------------+
| 4                 | 0.784             | 258.343           |
+-------------------+-------------------+-------------------+
| 5                 | 0.802             | 258.788           |
+-------------------+-------------------+-------------------+
| 6                 | 0.831             | 259.237           |
+-------------------+-------------------+-------------------+
| 7                 | 0.859             | 259.691           |
+-------------------+-------------------+-------------------+
| 8                 | 0.879             | 260.118           |
+-------------------+-------------------+-------------------+
| 9                 | 0.896             | 260.571           |
+-------------------+-------------------+-------------------+
| 10                | 0.910             | 261.044           |
+-------------------+-------------------+-------------------+
```
(TF performs better when passed pre-computed deep features - investigating this currently)

### CODE REVIEW

**Comment by fareeha** [comment]
Two tests breaking due to NAG optimizer. Restoring back to Adam gives:

**ESC-10 on MXNet 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.371             | 269.458           |
+-------------------+-------------------+-------------------+
| 2                 | 0.591             | 269.737           |
+-------------------+-------------------+-------------------+
| 3                 | 0.677             | 270.024           |
+-------------------+-------------------+-------------------+
| 4                 | 0.781             | 270.303           |
+-------------------+-------------------+-------------------+
| 5                 | 0.820             | 270.591           |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 270.877           |
+-------------------+-------------------+-------------------+
| 7                 | 0.864             | 271.173           |
+-------------------+-------------------+-------------------+
| 8                 | 0.886             | 271.505           |
+-------------------+-------------------+-------------------+
| 9                 | 0.900             | 271.973           |
+-------------------+-------------------+-------------------+
| 10                | 0.911             | 272.249           |
+-------------------+-------------------+-------------------+
```

**ESC-10 on TF 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.796             | 295.410           |
+-------------------+-------------------+-------------------+
| 2                 | 0.879             | 295.684           |
+-------------------+-------------------+-------------------+
| 3                 | 0.916             | 295.959           |
+-------------------+-------------------+-------------------+
| 4                 | 0.926             | 296.237           |
+-------------------+-------------------+-------------------+
| 5                 | 0.943             | 296.501           |
+-------------------+-------------------+-------------------+
| 6                 | 0.941             | 296.763           |
+-------------------+-------------------+-------------------+
| 7                 | 0.944             | 297.025           |
+-------------------+-------------------+-------------------+
| 8                 | 0.945             | 297.288           |
+-------------------+-------------------+-------------------+
| 9                 | 0.946             | 297.545           |
+-------------------+-------------------+-------------------+
| 10                | 0.946             | 297.815           |
+-------------------+-------------------+-------------------+
```

**ESC-50 on MXNet 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.197             | 599.795           |
+-------------------+-------------------+-------------------+
| 2                 | 0.356             | 600.804           |
+-------------------+-------------------+-------------------+
| 3                 | 0.508             | 601.785           |
+-------------------+-------------------+-------------------+
| 4                 | 0.621             | 602.775           |
+-------------------+-------------------+-------------------+
| 5                 | 0.691             | 603.766           |
+-------------------+-------------------+-------------------+
| 6                 | 0.747             | 604.763           |
+-------------------+-------------------+-------------------+
| 7                 | 0.789             | 605.758           |
+-------------------+-------------------+-------------------+
| 8                 | 0.816             | 606.746           |
+-------------------+-------------------+-------------------+
| 9                 | 0.839             | 607.725           |
+-------------------+-------------------+-------------------+
| 10                | 0.857             | 608.708           |
+-------------------+-------------------+-------------------+
```

**ESC-50 on TF 6.0**
```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.237             | 593.544           |
+-------------------+-------------------+-------------------+
| 2                 | 0.538             | 594.641           |
+-------------------+-------------------+-------------------+
| 3                 | 0.643             | 595.747           |
+-------------------+-------------------+-------------------+
| 4                 | 0.733             | 596.842           |
+-------------------+-------------------+-------------------+
| 5                 | 0.793             | 597.936           |
+-------------------+-------------------+-------------------+
| 6                 | 0.846             | 599.025           |
+-------------------+-------------------+-------------------+
| 7                 | 0.859             | 600.117           |
+-------------------+-------------------+-------------------+
| 8                 | 0.874             | 601.219           |
+-------------------+-------------------+-------------------+
| 9                 | 0.896             | 602.311           |
+-------------------+-------------------+-------------------+
| 10                | 0.902             | 603.394           |
+-------------------+-------------------+-------------------+
```
